### PR TITLE
Detect integers 6644 v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6644 and all subtickets
https://redmine.openinfosecfoundation.org/issues/6645
https://redmine.openinfosecfoundation.org/issues/6646
https://redmine.openinfosecfoundation.org/issues/6647
https://redmine.openinfosecfoundation.org/issues/6648
https://redmine.openinfosecfoundation.org/issues/6628

Describe changes:
- detect/integers: support hexadecimal notation for parsing
- detect/integers: add mode for negated range
- detect/integers: rust derive for enumerations
- detect/integers: keywords now accept bitmasks
- doc: detect/integers

https://github.com/OISF/suricata/pull/10105 with doc commit + improved derive cc @jasonish

### CODE REVIEW

**Comment by catenacyber** [comment]
cc @jufajardini for the doc commit ;-)

File: rust/src/detect/uint.rs [hunk 1]
```diff
@@ -46,6 +48,27 @@ pub struct DetectUintData<T> {
...
+    fn from(value: &dyn Enum<T>) -> Self {
+        return DetectUintData::<T> {
+            arg1: value.into_u(),
+            arg2: T::min_value(),
+            mode: DetectUintMode::DetectUintModeEqual,
+        };
+    }
+}
+
+pub fn detect_parse_uint_enum<T1: DetectIntType, T2: Enum<T1>>(s: &str) -> Option<DetectUintData<T1>> where DetectUintData<T1>: From<T2> {
```
Comment (catenacyber) [inline rust/src/detect/uint.rs:63]: @jasonish do you have a better way to just say `T2: Enum<T1: DetectIntType >`

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17354

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/10120

File: rust/src/detect/uint.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/detect/uint.rs:63]: This got modified in next version as I did not manage this generic+trait thing

**Pull request closed without merging**